src/Microsoft.OData.Edm/Csdl/CsdlXmlWriterSettings.cs

```diff
@@ -0,0 +1,19 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="CsdlXmlWriterSettings.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm.Csdl
+{
+    /// <summary>
+    /// Provides settings to be used with <see cref="CsdlXmlWriter"/>.
+    /// </summary>
+    public class CsdlXmlWriterSettings
+    {
+        /// <summary>
+        /// Gets or sets the compatibility flags for the <see cref="CsdlXmlWriter"/> to maintain compatibility with legacy functionality.
+        /// </summary>
+        public EdmLibraryCompatibility EdmLibraryCompatibility { get; set; }

```

Comment:
```suggestion
        public EdmLibraryCompatibility LibraryCompatibility { get; set; }
```
nit: To align naming with equivalent property in `ODataMessageWriterSettings`...

Reply:
They are not the same though..This has only Edm Library compatibility options.

Reply:
I didn't mean to rename the type. I meant to rename the property name. `CsdlXmlWriterSettings` and `EdmLibraryCompatibility` enum are types in the Edm library so naming the property `LibraryCompatibility` suffices, IMO.

Reply:
I think having them different will make it easy for someone to differentiate the EdmLibraryCompatibility and LibraryCompatibility properties. I could change it though.